packages/devtools_app/lib/src/screens/app_size/code_size_attribution.dart

```diff
@@ -193,13 +172,59 @@ class _CallGraphViewState extends State<CallGraphView> {
     );
   }
 
-  void _selectMainNode(CallGraphNode node) {
+  // TODO(kenz): store the selected node in a controller and pass the notifier
+  // to the tables instead of storing the [selectedNode] value in the state
+  // class.
+  void _selectMainNode(CallGraphNode? node) {
     setState(() {
-      selectedNode = node;
+      selectedNode = node!;
     });
   }
 }
 
+enum _CallGraphTableType {
+  from,
+  to,
+}
+
+class _CallGraphTable extends StatelessWidget {
+  const _CallGraphTable({
+    required this.tableType,
+    required this.selectedNode,
+    required this.onNodeSelected,
+  });
+
+  static const _fromDataKey = 'call-graph-from';
```

Comment:
Can you make these properties of `_CallGraphTableType` instead? That way we go from:

```dart
dataKey: tableType == _CallGraphTableType.from ? _fromDataKey : _toDataKey
```

To something like:

```dart
dataKey: tableType.dataKey
```